go/engine/signup.go

```diff
@@ -230,6 +230,9 @@ func (s *SignupEngine) registerDevice(m libkb.MetaContext, deviceName string) er
 	if err := RunEngine2(m, eng); err != nil {
 		return err
 	}
+	if err := eng.SetActiveDevice(m); err != nil {
+		return err
+	}
```

Comment:
sorry for another nit since i know this isn't your code, but can you change the signature of `storeSecret` below to not return an error? the function already just warns if it fails and we shouldn't have any errors once we commit to the new active device